system/jlib/jutil.cpp

```diff
@@ -2417,11 +2420,31 @@ jlib_decl bool querySecuritySettings(bool *          _useSSL,
                 configFileSpec.set(CONFIG_DIR).append(PATHSEPSTR).append("environment.conf");
 #endif
                 Owned<IProperties> conf = createProperties(configFileSpec.str(), true);
-                useSSL = conf->getPropBool("dfsUseSSL", false);
+                StringAttr sslMethod;
+                sslMethod.set(conf->queryProp("dfsUseSSL"));
+                if (sslMethod)
+                {
+                    // checking for true | false for backward compatibility
+                    if ( strieq(sslMethod.str(), "SSLOnly") || strieq(sslMethod.str(), "true") )
+                        useSSL = SSLOnly;
+                    else if ( strieq(sslMethod.str(), "SSLFirst") )
+                        useSSL = SSLFirst;
+                    else if ( strieq(sslMethod.str(), "LegacyFirst") )
+                        useSSL = LegacyFirst;
+                    else // SSLNone or false or ...
+                        useSSL = SSLNone;
+                }
                 if (useSSL)
                 {
                     certificate.set(conf->queryProp("dfsSSLCertFile"));
                     privateKey.set(conf->queryProp("dfsSSLPrivateKeyFile"));
+                    const char *passPhrasePtr = conf->queryProp("dfsSSLPassPhrase");
```

Comment:
I still am not clear how the encrypted password will get placed in environment.conf.  ConfigMgr encrypts passwords before writing to environment.xml, but env.conf is hand edited. Seems this leave password protected unusable?